File: 20_prime_num.c
//C program to check if entered number is prime or not

#include <stdio.h>

int main() {
    int num;

    printf("Enter any number: ");
    scanf("%d", &num);

    
    if(num<1){
        printf("%d is not a prime number.\n", num);
        return 0;
      }

    if(num==1){
      printf("1 is neither prime nor composite\n");
      return 0;
      }

    for(i = 2; i*i<=num; i++){
                if(num % i == 0){
                  flag = 1;
                  break;
        }
    }

    if (flag == 0){
        printf("%d is a prime number.\n", num);
    }
    else{
        printf("%d is not a prime number.\n", num);
    }

    return 0;
}

File: 20_prime_num.py
#Python program to check if a number is prime or not using function

import math

#design user-defined function to check fo prime
def is_prime(number):
    for i in range(int(math.sqrt(number))):
        if number%(i+1)==0:
            return False
        else:
            return True
            
            

#take input of number from user and check if prime or not            
num=int(input("Enter any integer: "))

result=is_prime(num)

if result==True:
    print("Entered number", num, " is a prime number")
else:
    print("Entered number ", num, "is not a prime number")


        
